openlcs/libs/sc_handler.py

```diff
@@ -0,0 +1,130 @@
+import koji
+import glob
+import os
+import uuid
+import sys
+
+# Fix absolute import issue in openlcs.
+openlcs_dir = os.path.abspath(os.path.join(os.path.dirname(__file__), '..'))
+if openlcs_dir not in sys.path:
+    sys.path.append(openlcs_dir)
+from libs.common import compress_source_to_tarball  # noqa: E402
+
+
+class SourceContainerHandler(object):
+    """
+    Object used for handle source in source containers.
+    """
+    def __init__(self, config=None):
+        self.config = config
+
+    @staticmethod
+    def get_component_flat(data, comp_type):
+        return {
+            'uuid': str(uuid.uuid4()),
+            'type': comp_type,
+            'name': data.get('name'),
+            'version': data.get('version'),
+            'release': data.get('release'),
+            'license': '',
+            'arch': '',
+            'synced': False
+        }
+
+    def get_container_component(self, sc_nvr):
+        """
+        Get container component.
+        """
+        nvr = koji.parse_NVR(sc_nvr)
+        container_component = self.get_component_flat(nvr, 'CONTAINER_IMAGE')
+        return container_component
+
+    def get_srpm_components(self, srpm_dir, sc_nvr):
+        """
+        Get the srpm components list from source container itself. The
+        component value is almost same as the srpm component from corgi.
+        The value of srpm component from corgi is as following:
+        {
+            'uuid': 'uuid',
+            'type': 'SRPM',
+            'name': 'libcom_err',
+            'version': '1.45.6',
+            'release': '2.el8',
+            'license': '',
+            'arch': '',
+            'synced': False
+        }
+        @params: srpm_dir, destination directory for srpm files' dir
+        """
+        components = []
+        if os.path.isdir(srpm_dir):
+            srpms = glob.glob(srpm_dir + "/*.src.rpm")
+            for srpm in srpms:
+                srpm = srpm.split('/')[-1]
+                rpm_nvr = srpm.replace('.src.rpm', '')
+                nvr = koji.parse_NVR(rpm_nvr)
```

Comment:
If directories contains only "*.src.rpm", you don't really need to split/replace like this, considering you'll also need "arch", try `koji.parse_NVRA` is much better and straight forward:

    In [1]: from koji import parse_NVRA
    In [2]: name = "libcom_err-1.45.6-2.el8.src.rpm"
    In [3]: parse_NVRA(name)
    Out[3]: 
    {'release': '2.el8',
     'version': '1.45.6',
     'name': 'libcom_err',
     'epoch': '',
     'arch': 'src',
     'src': True}

Reply:
Updated.